R/helper_add_fixed_drives.R

```diff
@@ -76,6 +76,19 @@ add_drive_results <- function(d) {
       ),
       # first observation of a half is also a new drive
       new_drive = dplyr::if_else(.data$row == 1, 1, .data$new_drive),
+
+      # if you recovered an onside kick or muffed return, it's a new drive
+      new_drive = dplyr::case_when(
+        .data$play_type == "kickoff" & (.data$own_kickoff_recovery == 1 | .data$fumble_lost == 1) ~ 1,
+        TRUE ~ .data$new_drive
+        ),
+
+      # if it's a kickoff and the prior play was a safety, it's a new drive
+      new_drive = dplyr::case_when(
+        .data$kickoff_attempt == 1 & dplyr::lag(.data$safety) == 1 ~ 1,
+        TRUE ~ .data$new_drive
+      ),
```

Comment:
Should we try to catch Timeouts between the safety and the kickoff?

Reply:
Done